sync-core/src/main/java/com/cloudant/sync/datastore/AttachmentManager.java

```diff
@@ -107,21 +107,35 @@ public void addAttachment(SQLDatabase db,PreparedAttachment a, BasicDocumentRevi
         values.put("revpos", revpos);
 
         // delete and insert in case there is already an attachment at this seq (eg copied over from a previous rev)
-        db.delete("attachments", " filename = ? and sequence = ? ", new String[]{filename, String.valueOf(sequence)});
+        db.delete("attachments", " filename = ? and sequence = ? ", new String[]{filename,
+                String.valueOf(sequence)});
         long result = db.insert("attachments", values);
         if (result == -1) {
             // if we can't insert into DB then don't copy the attachment
             a.tempFile.delete();
-            throw new SQLException("Could not insert attachment " + a + " into database with values " + values + "; not copying to attachments directory");
+            throw new AttachmentNotSavedException("Could not insert attachment " + a + " into database with values " + values + "; not copying to attachments directory");
         }
         // move file to blob store, with file name based on sha1
         File newFile = fileFromKey(sha1);
-        try{
-            FileUtils.moveFile(a.tempFile, newFile);
-        } catch (FileExistsException fee) {
-            // File with same SHA1 hash in the store, we assume it's the same content so can discard
-            // the duplicate data we have just downloaded
-            a.tempFile.delete();
+
+        if(newFile.exists()){
```

Comment:
I'm not sure if it's useful to add this newFile.exists() check. You're still catching the FileExistsException anyway, so it's redundant. Obviously you can't remove that catch because there would be a race condition on the file system. So just try to do the move and deal with it in the catch block.